### PR TITLE
Adding input validation before saving to the db

### PR DESCRIPTION
**Pull request by lslschr**
Adding input validation

### CODE REVIEW

File: endpoints/categories/category.php [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+    $value = trim($value);
+    $value = stripslashes($value);
+    $value = htmlspecialchars($value);
+    $value = htmlentities($value);
+    return $value;
+}
 if (isset($_SESSION['loggedin']) && $_SESSION['loggedin'] === true) {
     if (isset($_GET['action']) && $_GET['action'] == "add") {
-        $categoryName = "Category";
+        $categoryName = validate("Category");
```
Comment (ellite) [inline endpoints/categories/category.php:13]: This value "Category" is hardcoded and not a user input, no need for validation here.

File: endpoints/currency/currency.php [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+    $value = htmlspecialchars($value);
+    $value = htmlentities($value);
+    return $value;
+}
 if (isset($_SESSION['loggedin']) && $_SESSION['loggedin'] === true) {
     if (isset($_GET['action']) && $_GET['action'] == "add") {
-        $currencyName = "Currency";
-        $currencySymbol = "$";
-        $currencyCode = "CODE";
+        $currencyName = validate("Currency");
```
Comment (ellite) [inline endpoints/currency/currency.php:13]: Same on this three.

File: endpoints/household/household.php [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+    $value = trim($value);
+    $value = stripslashes($value);
+    $value = htmlspecialchars($value);
+    $value = htmlentities($value);
+    return $value;
+}
 if (isset($_SESSION['loggedin']) && $_SESSION['loggedin'] === true) {
     if (isset($_GET['action']) && $_GET['action'] == "add") {
-        $householdName = "Member";
+        $householdName = validate("Member");
```
Comment (ellite) [inline endpoints/household/household.php:13]: Same here.

**Review by ellite — CHANGES_REQUESTED** [review]
Thank you.
I left some comments.

**Comment by lslschr** [comment]
Hi 
Thank you very much for the feedback. 

Just removed the unnecessary validations.

**Review by ellite — APPROVED** [review]
Thank you.

**Comment by lslschr** [comment]
Just wanted to mention the reason for this PR: 
The application wallos in the version 0.9 is vulnerable to Cross-Site Scripting (XSS) affecting all text-based input fields across the application, where input validation is not properly implemented. To exploit this vulnerability, the attacker can insert the following payload into a text-based input field: <script>alert("xss")</script>. Once saved, refreshing the browser will trigger the execution of this JavaScript code in the user's browser. Similarly, the script will execute when a different user accesses the application and encounters the injected payload.

So the vulnerability is a XSS vulnerability at all text-based input fields and the root cause for this vulnerability is the missing input validation or the incorrect implementation of these. This input validation was created as part of this PR and the vulnerability was therefore closed. 

The XSS vulnerability affects the security of users who use the application. Other users will experience the same vulnerability when using the application. XSS can cause a variety of problems for the end user that range in severity from an annoyance to complete account compromise. The most severe XSS attacks involve disclosure of the user’s session cookie, allowing an attacker to hijack the user’s session and take over the account.

**Pull request merged**